### ISSUE TITLE
Missing "Remove" button for messages in upgraded rooms seems like major privacy flaw

### ISSUE THREAD

**Issue by ell1e** [vector-im/element-web#23446]
### Steps to reproduce

The missing "Remove" button for messages in the old copies of upgraded rooms seems like major privacy flaw given the dumping of all "anyone" history rooms on the `matrix.org` homeserver into the google index via `view.matrix.org`. Since not only will most people be unaware they're being indexed to that intrusive level, once they find out they can't even remove their contents from the index again since the old copies of upgraded rooms seem to remain indexed fully with all contents on `view.matrix.org`, but now suddenly it's just stuck there and unremovable. I think that's pretty bad.

1. Speak in a room with history visibility "anyone"
2. Room gets upgraded
3. Find your message in the google index and that there is no way to delete it

#### What did you expect?
I can still delete the messages. Also, preferably Element would show some sort of symbol in some corner indicating all messages end up on a publicly indexed site in full.

#### What happened instead?
Messages remain removable in a way that also removes them from `view.matrix.org`. Also, there still seems to be zero notification of any kind when joining such a room that all contents are dumped straight into the Google index, which given how Discord, Telegram, etc. don't do that  I would say is hugely unexpected.

(Just as a reminder, that you could in theory join and get the history anyway is pretty irrelevant, I discussed this on the other ticket. Exposure, searchability, and low barriers to abuse matter for privacy, not just the theoretical ability to dig in someone's stuff, a threat model of "it's theoretically out there anyway" isn't useful for real world impact.)

### Operating system

Linux/Fedora

### Application version

 1.11.8

### How did you install the app?

flatpak from flathub

### Homeserver

matrix.org

### Will you send logs?

No

**Comment by t3chguy** [comment]
This is because the room's permissions got changed such that you can no longer redact by the server when the upgrade API got called. It updates `events_default` which also governs `m.room.redaction` events

![image](https://user-images.githubusercontent.com/2403652/194842335-4a1e9647-8661-4f19-905f-d049cd67fbb5.png)

**Comment by ell1e** [comment]
Okay but that just gives a technical background, it doesn't justify why it is like that.

I think again this is a major privacy issue and should be fixed. Obviously I couldn't tell you how, but maybe the backend developers have an idea?

**Comment by ell1e** [comment]
There is also the problem that after clicking the upgrade button the old room view gets removed, and even if I still had delete permissions how could a user even find the messages again to see them?

The whole process seems poorly thought out from a privacy angle.

**Comment by t3chguy** [comment]
> There is also the problem that after clicking the upgrade button the old room view gets removed, and even if I still had delete permissions how could a user even find the messages again to see & delete them?

Either scroll to the top of the new room to the link to the old one or in Settings there is also a link to the older room

**Comment by ell1e** [comment]
> Either scroll to the top of the new room to the link to the old one or in Settings there is also a link to the older room

That makes me wonder, will search even cross that boundary though? Because am I otherwise expected to somehow get to the top of maybe hundreds of pages of logs just to find that link if I want really *all* messages in the search of a specific person? (Assuming the search ever gets fixed at all such that searching for people's messages works as in https://github.com/vector-im/element-web/issues/23447 and https://github.com/matrix-org/synapse/issues/14121 )

**Comment by t3chguy** [comment]
> will search even cross that boundary though? 

Yup, it should. Seshat and Synapse at least do, though I don't see it codified in the spec.
